sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/FcdaRecordCsvHelper.java

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: 2023 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.util;
+
+import com.opencsv.bean.CsvBindByPosition;
+import lombok.Getter;
+import lombok.ToString;
+import org.lfenergy.compas.scl2007b4.model.TFCEnum;
+
+import java.io.Reader;
+import java.util.List;
+
+/**
+ * This class is a helper method to load FcdaRecord from a CSV file for use with {@link org.lfenergy.compas.sct.commons.scl.HmiService#createAllHmiReportControlBlocks}
+ * Use the getter getFcdaRecords to access the list of parsed FcdaRecord.
+ *
+ * @see CsvUtils
+ * @see org.lfenergy.compas.sct.commons.scl.HmiService#createAllHmiReportControlBlocks
+ */
+public class FcdaRecordCsvHelper {
+
+    @Getter
+    private final List<FcdaRecord> fcdaRecords;
+
+    /**
+     * Constructor
+     * Provide the CSV file as a Reader. For example, you can create a reader like this :
+     * <code>new InputStreamReader(getClass().getClassLoader().getResourceAsStream(fileName), StandardCharsets.UTF_8);</code>
+     *
+     * @param csvSource a reader that provides the data as CSV
+     */
+    public FcdaRecordCsvHelper(Reader csvSource) {
+        fcdaRecords = CsvUtils.parseRows(csvSource, Row.class).stream()
```

Comment:
I don't want to expose my `Row` inner class, which has openscv annotations, to a public compas service. I prefer to use a DTO. Where I can improve though, is that I can get rid of FcdaRecord from production code and use the TFCDA class (generated by Jaxb) directly.

Reply:
I don't catch the difference

Reply:
OK, since you're now remapping the CSV class to an Xml Class.  
The idea was to use an object with the required types wich is not the same that the one mapped right after.  
Here you can rename `Row` in `FdcaRecord` by the way.  
And use an enum directly for `fc` colomn :  
```java
@CsvCustomBindByPosition(position = 5)
        private TFCEnum fc;
```
https://opencsv.sourceforge.net/#enumerations  

Finally `SclConstructorHelper.newFcda` seems a bit useless ...
```java
.map(row -> {
                            TFCDA tfcda = new TFCDA();
                            tfcda.setLdInst(row.ldInst);
                            if (StringUtils.isNotBlank(row.lnClass)) tfcda.getLnClass().add(row.lnClass);
                            tfcda.setLnInst(row.lnInst);
                            tfcda.setPrefix(StringUtils.trimToEmpty(row.prefix)); // TFCDA prefix is never null
                            tfcda.setDoName(row.doName);
                            tfcda.setFc(TFCEnum.valueOf(row.fc));
                            return tfcda;
                        }
                )
```
or use dedicated mapper inside the class : 
```java
.map(row -> toTfcda(row))
```
```java
private static TFCDA toTfcda(Row row) {
        TFCDA tfcda = new TFCDA();
        tfcda.setLdInst(row.ldInst);
        if (StringUtils.isNotBlank(row.lnClass)) tfcda.getLnClass().add(row.lnClass);
        tfcda.setLnInst(row.lnInst);
        tfcda.setPrefix(StringUtils.trimToEmpty(row.prefix)); // TFCDA prefix is never null
        tfcda.setDoName(row.doName);
        tfcda.setFc(TFCEnum.valueOf(row.fc));
        return tfcda;
    }
```

Reply:
the PR is approuved by the way

Reply:
I did not know OpenCsv can convert to enum directly. I changed it.

I use `SclConstructorHelper.newFcda` in DataSetAdapter#createFCDAIfNotExists, so I keep it.